BuildTasks/Common/Common.ts

```diff
@@ -150,6 +150,26 @@ export function runTfx(cmd: (tfx: ToolRunner) => void) {
 }
 
 /**
+ * Reads the extension version from the supplied variable
+ *
+ * @param  {string="connectedServiceName"} inputFieldName
+ * @returns string
+ */
+export function getExtensionVersion(): string {
+    const extensionVersion = tl.getInput("extensionVersion", false);
+    if (extensionVersion) {
+        const extractedVersions = extensionVersion.match(/[0-9]+\.[0-9]+\.[0-9]+(\.[0-9]+)?/);
```

Comment:
We need to use a non-capturing group for the 4th part, otherwise, we will get 2 matches, which then the if clause will fail. I can change it before merge if you prefer.